hydra-node/src/Hydra/Network/Reliability.hs

```diff
@@ -124,6 +130,44 @@ instance Arbitrary (Vector Int) where
 instance Arbitrary ReliabilityLog where
   arbitrary = genericArbitrary
 
+-- | Handle for all persistence operations in the Reliability network layer.
+-- This handle takes care of storing and retreiving vector clock and all
+-- messages.
+data MessagePersistence m msg = MessagePersistence
+  { loadAcks :: m (Vector Int)
+  , saveAcks :: Vector Int -> m ()
+  , loadMessages :: m [Heartbeat msg]
+  , appendMessage :: Heartbeat msg -> m ()
+  }
+
+-- | Create 'MessagePersistence' out of 'PersistenceIncremental' and
+-- 'Persistence' handles. This handle loads and saves acks (vector clock data)
+-- and can load and append network messages.
+-- On start we construct empty ack vector from all parties in case nothing
+-- is stored on disk.
+-- NOTE: This handle is returned in the underlying context just for the sake of
+-- convenience.
+mkMessagePersistence ::
+  (MonadThrow m, FromJSON msg, ToJSON msg) =>
+  Vector a ->
+  PersistenceIncremental (Heartbeat msg) m ->
+  Persistence (Vector Int) m ->
+  MessagePersistence m msg
+mkMessagePersistence allParties msgPersistence ackPersistence =
```

Comment:
We don't need `allParties`, just the lenght, which removes the need to provide a `Vector`